### ISSUE TITLE
Reduce CI setup time with binary releases

### ISSUE THREAD

**Issue by epage** [roblabla/cargo-travis#11]
Instead of `cargo install cargo-travis`, it'd be helpful to have pre-compiled releases.

For example, you can do the following to install rustfmt:
```curl -LSfs https://japaric.github.io/trust/install.sh | sh -s -- --force --crate rustfmt --git japaric/rustfmt-bin --tag v0.6.3-20161120```

**Comment by roblabla** [comment]
Yup. I'll probably start using trust down the line for CI testing and setting up binary releases.

**Comment by iwburns** [comment]
While I know this doesn't solve your problem 100%, I'm currently using this to speed things up:
`which cargo-coverage && which cargo-coveralls || cargo install cargo-travis`.

**Comment by Geal** [comment]
Hey,
I use more or less the same command, but kcov still gets built: https://travis-ci.org/sozu-proxy/sozu/jobs/316438026#L1159
Is it linked to #6? How can I help?
(not trying to pressure you to fix this quickly, I'd like a faster build, but I can live without it)

**Comment by roblabla** [comment]
Right now, `kcov` gets built at every build, inside `CWD/kcov/build/src/kcov`. This is problematic for two reasons: It causes #6 (CWD is different for each member of the workspace), and that folder is not cached. 

According to the travis docs, when using `cache: cargo`, `$HOME/.cargo` and `$TRAVIS_BUILD_DIR/target` get cached.

I think the proper fix would be to build kcov into `$TRAVIS_BUILD_DIR/target/kcov` or something, giving it a global location (Fixing #6) and allowing it to get cached (fixing your issue). 

The procedure to locate and/or build kcov is in the [build_kcov function](https://github.com/roblabla/cargo-travis/blob/master/src/lib.rs#L110). Ideally, we should make the script run in the target folder, which we could simply pass as an argument to that function, obtaining that function from the Workspace instance [here](https://github.com/roblabla/cargo-travis/blob/master/src/bin/cargo-coveralls.rs#L94)

**Comment by epage** [comment]
Regarding workarounds

> `which cargo-coverage && which cargo-coveralls || cargo install cargo-travis`

I eventually went with this but it was a pain for upgrades.

For cargo-when, I went with
`if [[ `cargo-when --version` != *$WHEN ]] ; then travis_wait cargo install cargo-when --force --vers $WHEN; fi`

I'd love to go with that but I didn't find an easily parseable version for cargo-travisreport the version

**Comment by roblabla** [comment]
Adding a `--version` flag for cargo-travis should be fairly simple. I can do that ASAP.

**Comment by roblabla** [comment]
@epage latest version prints "cargo-travis 0.0.7" when you run `cargo-coverage -V`. You can get just the version with a `cargo coverage -V | cut -d' ' -f2`

**Comment by epage** [comment]
Thanks!
